grid-proxy/internal/explorer/db/postgres.go

```diff
@@ -76,6 +76,35 @@ func (d *PostgresDatabase) Close() error {
 	return db.Close()
 }
 
+func (d *PostgresDatabase) Ping() error {
+	db, err := d.gormDB.DB()
+	if err != nil {
+		return fmt.Errorf("failed to get db connection")
+	}
+
+	if err := db.Ping(); err != nil {
+		return fmt.Errorf("failed to ping db")
+	}
+
+	return nil
+}
+
+func (d *PostgresDatabase) Initialized() error {
+	db, err := d.gormDB.DB()
+	if err != nil {
+		return fmt.Errorf("failed to get db connection")
+	}
+
+	initTables := []string{"node_gpu", "resources_cache"}
+	for _, tableName := range initTables {
```

Comment:
we can just check if the table exists without loading all the data with the select statement 
something like that 
```
SELECT EXISTS (
   SELECT 1
   FROM pg_tables
   WHERE schemaname = 'public' // change to correct schema
   AND tablename = tableName
);
```